builds/jenkins-delete-resources/Jenkinsfile

```diff
@@ -0,0 +1,395 @@
+import groovy.json.JsonSlurper
+import groovy.transform.Field
+import groovy.json.JsonOutput
+
+@Field def inputParams
+@Field def config
+@Field def awsAccounts
+@Field def finalAccounts = []
+@Field def credentialId
+@Field def isPrimary = false
+
+
+// To be replaced as @Field def repo_credential_id = "value" for repo_credential_id, repo_base and repo_core
+@Field def repo_credential_id 
+@Field def aws_credential_id 
+@Field def region 
+@Field def instance_prefix 
+@Field def repo_base 
+@Field def repo_core
+@Field def scm_type 
+
+@Field def configModule
+@Field def configLoader
+
+
+node {
+ echo "Starting the Job with Params ${params}"
+ inputParams = params.input
+ // Load the config
+ def jazzBuildModuleURL = getBuildModuleUrl()
+ loadBuildModules(jazzBuildModuleURL)
+ def output = JsonOutput.toJson(configLoader)
+ echo "${output}"
+ //check whether input param is not null
+
+ if (inputParams == "") {
+  echo "No Params defined"
+ } else if (inputParams == "all") {
+  finalAccounts = configLoader.AWS['ACCOUNTS']
+ } else {
+  // Split the Input parameter by comma seperated value
+  String[] inputstr;
+  inputstr = inputParams.split(',');
+  awsAccounts = configLoader.AWS['ACCOUNTS']
+  // Filter the List of Accounts to delete
+  for (awsAccount in awsAccounts) {
+   // Check the AccountId is in inputStr array
+   def accountId = awsAccount['ACCOUNTID']
+   if (inputstr.contains(accountId)) {
+    finalAccounts.push(awsAccount)
+   }
+  }
+ }
+
+ if (finalAccounts.size() > 0) {
+  //Parse the Accounts and delete IAM,APIGateway,s3Buckets
+  for (account in finalAccounts) {
+   credentialId = account['CREDENTIAL_ID']
+   def iamRoles = account["IAM"]
+   isPrimary = account['PRIMARY']
+   if(!isPrimary){
+    withCredentials([
+      [$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "${credentialId}"]
+    ]) {
+      // check if iamRoles are not empty
+      if (iamRoles.size() > 0) {
+      // start the deletion process of IAM Roles
+      processIamRoles(iamRoles)
+      } else {
+      // Nothing to do, Just log the message
+      echo "No IAM Roles found for deletion"
+      }
+      // Get all Regions
+      def allregions = account["REGIONS"]
+      for (awsregion in allregions) {
+      echo "processing  Region ${awsregion['REGION']}"
+      def regionAws = awsregion['REGION']
+      //Get Region based apiGateway and s3Buckets
+      def apiGateway = awsregion["API_GATEWAY"]
+      def s3Buckets = awsregion["S3"]
+      def cloudFront = awsregion["CLOUDFRONT"]
+      // Check ApiGateway is not Null
+      if (apiGateway.size() > 0) {
+        // Trigger Deletion process of APIGateway
+        processApiGateway(apiGateway, regionAws)
```

Comment:
Do the checks in side particular methods, so that code will look more cleaner

Reply:
I guess this check is important, as it checks the size before entering on to the function.